heat/core/communication.py

```diff
@@ -1401,20 +1402,21 @@ def Alltoall(
 
     def Alltoallv(
         self,
-        sendbuf: Union[DNDarray, torch.Tensor, Any],
-        recvbuf: Union[DNDarray, torch.Tensor, Any],
+        sendbuf: Tuple[DNDarray, torch.Tensor, Any],
+        recvbuf: Tuple[DNDarray, torch.Tensor, Any],
```

Comment:
thanks @swaptr , you're actually validating my impression that the documentation is unclear, so thanks a lot for tackling this.

For Alltoallv, `sendbuf` and `recvbuf` must be a tuple of: data object, counts, displacements. Check out the [official OpenMPI documentation](https://www.open-mpi.org/doc/v4.0/man3/MPI_Alltoallv.3.php). Here's another [explanation](https://rookiehpc.github.io/mpi/docs/mpi_alltoallv/index.html) I get back to often as it's easier to understand.

Here's an [example in our `ht.reshape()` function](https://github.com/helmholtz-analytics/heat/blob/e5e8a96c28c0508df4d2865a816e40e74ac4491c/heat/core/manipulations.py#L1971)